applications/call_inspector/src/parsers/ci_chunk.erl

```diff
@@ -0,0 +1,295 @@
+%%%-------------------------------------------------------------------
+%%% @copyright (C) 2012-2014, 2600Hz INC
+%%% @doc
+%%%
+%%% @end
+%%% @contributors
+%%%
+%%%-------------------------------------------------------------------
+-module(ci_chunk).
+
+-export([new/0]).
+-export([setters/2]).
+-export([call_id/2, call_id/1]).
+-export([append_data/2
+         ,data/2
+         ,data/1
+        ]).
+-export([timestamp/2, timestamp/1]).
+-export([ref_timestamp/1]).
+-export([dst_ip/2, dst_ip/1]).
+-export([dst_port/2, dst_port/1]).
+-export([src_ip/2, src_ip/1]).
+-export([src_port/2, src_port/1]).
+-export([parser/2, parser/1]).
+-export([label/2, label/1]).
+-export([to_json/1
+         ,from_json/1
+        ]).
+-export([is_chunk/1]).
+-export([sort_by_timestamp/1
+         ,reorder_dialog/1
+        ]).
+
+-record(ci_chunk, {call_id :: ne_binary()
+                  ,data = [] :: ne_binaries()
+                  ,timestamp :: number()
+                  ,ref_timestamp :: number()
+                  ,src_ip :: ne_binary()
+                  ,src_port :: pos_integer()
+                  ,dst_ip :: ne_binary()
+                  ,dst_port :: pos_integer()
+                  ,parser :: atom()
+                  ,label :: ne_binary()
+                 }).
+-type chunk() :: #ci_chunk{}.
+
+-export_type([chunk/0]).
+
+-include("../call_inspector.hrl").
+
+-define(setter(Field),
```

Comment:
We've conventionally UPPERCASED macros to set them apart visually. Unless others disagree, I'd like to continue doing so.